marconi-cardano-indexers/src/Marconi/Cardano/Indexers/BlockInfo.hs

```diff
@@ -239,52 +232,104 @@ instance
       (\cp -> pure [":slotNo" := C.chainPointToSlotNo cp])
       (const blockInfoBySlotNoQuery)
       (const listToMaybe)
+  queryLatest =
+    Core.queryLatestSQLiteIndexerWith
+      (pure [])
+      (const lastBlockInfoQuery)
+      (const listToMaybe)
+
+queryClosestBlockInfo
+  :: ( MonadIO m
+     , MonadError (Core.QueryError (Core.EventsMatchingQuery BlockInfo)) m
+     )
+  => Maybe C.ChainPoint
+  -> Core.EventsMatchingQuery BlockInfo
+  -> Core.SQLiteIndexer BlockInfo
+  -> m (Core.Result (Core.EventsMatchingQuery BlockInfo))
+queryClosestBlockInfo =
+  let queryPrefix :: SQL.Query
+      queryPrefix =
+        [sql|
+        SELECT blockNo, blockTimestamp, epochNo,
+               slotNo, blockHeaderHash
+        FROM blockInfo
+        |]
+      queryLatest :: SQL.Query
+      queryLatest = queryPrefix
+      querySpecific :: SQL.Query
+      querySpecific = queryPrefix <> [sql| WHERE slotNo <= :slotNo |]
+      parseResult
+        :: (a -> Maybe a)
+        -> [Core.Timed C.ChainPoint a]
+        -> [Core.Timed C.ChainPoint a]
+      parseResult = mapMaybe . traverse
+   in \case
+        Nothing ->
+          Core.queryLatestSQLiteIndexerWith
+            (pure [])
+            (const queryLatest)
```

Comment:
`queryLatest` here can be replaced with `queryPrefix` and the alias in the `let` deleted.